numpy/_core/src/multiarray/methods.c

```diff
@@ -962,12 +963,18 @@ array_getarray(PyArrayObject *self, PyObject *args, PyObject *kwds)
     if (copy == NPY_COPY_ALWAYS) {
         if (newtype == NULL) {
             newtype = PyArray_DESCR(self);
+            // Take a new strong reference to compensate for
+            // PyArray_CastToType stealing a reference to newtype.
+            Py_INCREF(newtype);
```

Comment:
```suggestion
            Py_INCREF(newtype);  // newtype is owned.
```
Maybe. I don't like the overly long comments, TBH.

We could add a comment on all `CastToType` calls that it steals the reference, though.